Rolling upgrade test failure related to ESQL

### CI Link

https://gradle-enterprise.elastic.co/s/t3fre2k6cszyc

### Repro line

`gradlew :x-pack:qa:rolling-upgrade:v8.10.0#twoThirdsUpgradedTest -Dtests.class="org.elasticsearch.upgrades.UpgradeClusterClientYamlTestSuiteIT" -Dtests.method="test {p0=mixed_cluster/90_ml_data_frame_analytics_crud/Get old classification job stats}" -Dtests.seed=D34AC6B8F4DE1F2F -Dtests.bwc=true -Dtests.locale=es-GT -Dtests.timezone=Africa/El_Aaiun -Druntime.java=20`

### Does it reproduce?

Yes

### Applicable branches

main

### Failure history

_No response_

### Failure excerpt

This fails on my machine but in a different way.  What I noticed in the CI logs though is an error related to ESQL that _might_ be the reason for the failure, but I didn't dig into specifics:

```
» [2023-08-17T13:33:07,928][WARN ][o.e.t.TcpTransport       ] [v8.10.0-2] exception caught on transport layer [Netty4TcpChannel{localAddress=/127.0.0.1:42469, remoteAddress=/127.0.0.1:54238, profile=default}], closing connection java.lang.Exception: java.lang.AssertionError: cluster:monitor/xpack/esql/stats/dist[n]
»  	at org.elasticsearch.transport.netty4@8.10.0-SNAPSHOT/org.elasticsearch.transport.netty4.Netty4MessageInboundHandler.exceptionCaught(Netty4MessageInboundHandler.java:74)
»  	at io.netty.transport@4.1.94.Final/io.netty.channel.AbstractChannelHandlerContext.invokeExceptionCaught(AbstractChannelHandlerContext.java:346)
»  	at io.netty.transport@4.1.94.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:447)
»  	at io.netty.transport@4.1.94.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
»  	at io.netty.transport@4.1.94.Final/io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
»  	at io.netty.codec@4.1.94.Final/io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103)
»  	at io.netty.transport@4.1.94.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
»  	at io.netty.transport@4.1.94.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
»  	at io.netty.transport@4.1.94.Final/io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
»  	at io.netty.handler@4.1.94.Final/io.netty.handler.ssl.SslHandler.unwrap(SslHandler.java:1383)
»  	at io.netty.handler@4.1.94.Final/io.netty.handler.ssl.SslHandler.decodeJdkCompatible(SslHandler.java:1246)
»  	at io.netty.handler@4.1.94.Final/io.netty.handler.ssl.SslHandler.decode(SslHandler.java:1295)
»  	at io.netty.codec@4.1.94.Final/io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection(ByteToMessageDecoder.java:529)
»  	at io.netty.codec@4.1.94.Final/io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:468)
»  	at io.netty.codec@4.1.94.Final/io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:290)
»  	at io.netty.transport@4.1.94.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
»  	at io.netty.transport@4.1.94.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
»  	at io.netty.transport@4.1.94.Final/io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
»  	at io.netty.transport@4.1.94.Final/io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410)
»  	at io.netty.transport@4.1.94.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:440)
»  	at io.netty.transport@4.1.94.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
»  	at io.netty.transport@4.1.94.Final/io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919)
»  	at io.netty.transport@4.1.94.Final/io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:166)
»  	at io.netty.transport@4.1.94.Final/io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:788)
»  	at io.netty.transport@4.1.94.Final/io.netty.channel.nio.NioEventLoop.processSelectedKeysPlain(NioEventLoop.java:689)
»  	at io.netty.transport@4.1.94.Final/io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:652)
»  	at io.netty.transport@4.1.94.Final/io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562)
»  	at io.netty.common@4.1.94.Final/io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
»  	at io.netty.common@4.1.94.Final/io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
»  	at java.base/java.lang.Thread.run(Thread.java:1623)
»  Caused by: java.lang.AssertionError: cluster:monitor/xpack/esql/stats/dist[n]
»  	at org.elasticsearch.server@8.10.0-SNAPSHOT/org.elasticsearch.transport.InboundAggregator.lambda$new$0(InboundAggregator.java:46)
»  	at org.elasticsearch.server@8.10.0-SNAPSHOT/org.elasticsearch.transport.InboundAggregator.initializeRequestState(InboundAggregator.java:197)
»  	at org.elasticsearch.server@8.10.0-SNAPSHOT/org.elasticsearch.transport.InboundAggregator.headerReceived(InboundAggregator.java:66)
»  	at org.elasticsearch.server@8.10.0-SNAPSHOT/org.elasticsearch.transport.InboundPipeline.forwardFragments(InboundPipeline.java:136)
»  	at org.elasticsearch.server@8.10.0-SNAPSHOT/org.elasticsearch.transport.InboundPipeline.doHandleBytes(InboundPipeline.java:119)
»  	at org.elasticsearch.server@8.10.0-SNAPSHOT/org.elasticsearch.transport.InboundPipeline.handleBytes(InboundPipeline.java:84)
»  	at org.elasticsearch.transport.netty4@8.10.0-SNAPSHOT/org.elasticsearch.transport.netty4.Netty4MessageInboundHandler.channelRead(Netty4MessageInboundHandler.java:62)
»  	at io.netty.transport@4.1.94.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
»  	... 27 more
```

--- a/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plugin/TransportEsqlStatsAction.java
+++ b/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plugin/TransportEsqlStatsAction.java
@@ -6,9 +6,11 @@
  */
 package org.elasticsearch.xpack.esql.plugin;
 
+import org.elasticsearch.Version;
 import org.elasticsearch.action.FailedNodeException;
 import org.elasticsearch.action.support.ActionFilters;
 import org.elasticsearch.action.support.nodes.TransportNodesAction;
+import org.elasticsearch.cluster.ClusterState;
 import org.elasticsearch.cluster.node.DiscoveryNode;
 import org.elasticsearch.cluster.service.ClusterService;
 import org.elasticsearch.common.inject.Inject;
@@ -19,6 +21,7 @@ import org.elasticsearch.transport.TransportService;
 import org.elasticsearch.xpack.esql.execution.PlanExecutor;
 
 import java.io.IOException;
+import java.util.Arrays;
 import java.util.List;
 
 /**
@@ -52,6 +55,16 @@ public class TransportEsqlStatsAction extends TransportNodesAction<
             ThreadPool.Names.MANAGEMENT
         );
         this.planExecutor = planExecutor;
+    }
+
+    @Override
+    protected void resolveRequest(EsqlStatsRequest request, ClusterState clusterState) {
+        String[] nodesIds = clusterState.nodes().resolveNodes(request.nodesIds());
+        DiscoveryNode[] concreteNodes = Arrays.stream(nodesIds)
+            .map(clusterState.nodes()::get)
+            .filter(n -> n.getVersion().onOrAfter(Version.V_8_11_0))
+            .toArray(DiscoveryNode[]::new);
+        request.setConcreteNodes(concreteNodes);
     }
 
     @Override

